---
description: 编译器错误 CS1626
title: 编译器错误 CS1626
ms.date: 07/20/2015
f1_keywords:
- CS1626
helpviewer_keywords:
- CS1626
ms.assetid: 3ba03383-eb24-4fd8-bf40-8b0f7d6baf0d
ms.openlocfilehash: 234acba50df406bb6ea02cc5b5a8fe0c959027d5
ms.sourcegitcommit: d579fb5e4b46745fd0f1f8874c94c6469ce58604
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 08/30/2020
ms.locfileid: "89142500"
---
# <a name="compiler-error-cs1626"></a><span data-ttu-id="76117-103">编译器错误 CS1626</span><span class="sxs-lookup"><span data-stu-id="76117-103">Compiler Error CS1626</span></span>
<span data-ttu-id="76117-104">无法在包含 catch 子句的 Try 块体中生成值</span><span class="sxs-lookup"><span data-stu-id="76117-104">Cannot yield a value in the body of a try block with a catch clause</span></span>  
  
 <span data-ttu-id="76117-105">如果有一个 catch 子句与 try 块关联，该 try 块内不允许有 yield 语句。</span><span class="sxs-lookup"><span data-stu-id="76117-105">A yield statement is not allowed in a try block if there is a catch clause associated with the try block.</span></span> <span data-ttu-id="76117-106">若要避免此错误，请将 yield 语句移出 try/catch/finally 块，或删除 catch 块。</span><span class="sxs-lookup"><span data-stu-id="76117-106">To avoid this error, either move the yield statement out of the try/catch/finally block, or remove the catch block.</span></span>
  
 <span data-ttu-id="76117-107">下面的示例生成 CS1626：</span><span class="sxs-lookup"><span data-stu-id="76117-107">The following sample generates CS1626:</span></span>  
  
```csharp  
// CS1626.cs  
using System.Collections;  
  
class C : IEnumerable  
{  
   public IEnumerator GetEnumerator()  
   {  
      try  
      {  
         yield return this;  // CS1626  
      }  
      catch  
      {  
  
      }  
      finally
      {

      }
   }  
}  
  
public class CMain  
{  
   public static void Main() { }  
}  
```
